Lib/test/test_tcl.py

```diff
@@ -142,7 +142,9 @@ def test_getint(self):
         for i in self.get_integers():
             self.assertEqual(tcl.getint(' %d ' % i), i)
             self.assertEqual(tcl.getint(' %#o ' % i), i)
-            self.assertEqual(tcl.getint((' %#o ' % i).replace('o', '')), i)
+            self.assertEqual(tcl.getint((' %#o ' % i).replace('o', '')),
+                             i if tcl_version < (9, 0) else
+                             int(('%#o' % i).replace('o', '')))
```

Comment:
```suggestion
            self.assertEqual(tcl.getint(' 0%o ' % i),
                             i if tcl_version < (9, 0) else int(' 0%o ' % i))
```

Reply:
The purpose of of this test is testing difference between `getint()` and `int()`, i.e. testing testing that `getint()` uses Tcl's code instead of calling `int()`.

`' 0%o ' % i` does not work for negative numbers.